Makefile

```diff
@@ -4,6 +4,7 @@ LDFLAGS = -X github.com/trento-project/agent/version.Version="$(VERSION)"
 LDFLAGS := $(LDFLAGS) -X github.com/trento-project/agent/version.InstallationSource="$(INSTALLATIONSOURCE)"
 ARCHS ?= amd64 arm64 ppc64le s390x
 DEBUG ?= 0
+BUILD_DIR ?= ./build
```

Comment:
This new default folder looks dangerous.
Many of our package generation pipelines (rpm creation spec files for example) depend on having the `trento-agent` binary in the root folder

Reply:
I'll try to fix everything to have all the artifacts under `./build`, it seems more tidy to me. If I cannot, I just revert to the current behavior

Reply:
In the end, the release pipelines do not rely on the `make build` command, but use the `make cross-compile` instead, which creates an artifact for each supported architecture; The artifacts are placed into `build/<arch>` folders. 

Thanks to @CDimonaco's suggestion I opted for using the same convention for the development build and test.
So `make build` creates the artifact into the `build/${GOARCH}` folder, depending on the current system; so does `make build-plugin-examples` as well.

I think it's safe now.

Reply:
@balanza The `rpm` creation does depend on the `make build` command and the output it generates: https://github.com/trento-project/agent/blob/main/packaging/suse/trento-agent.spec#L52

Reply:
You are right, I forgot that part. I propose this to make it compatible: https://github.com/trento-project/agent/pull/362/commits/7e5320638f6568dde8f535a8447d5ef499685e0a